internal/credit/postgres_connector/watermark.go

```diff
@@ -0,0 +1,44 @@
+package postgres_connector
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"entgo.io/ent/dialect/sql"
+
+	"github.com/openmeterio/openmeter/internal/credit/postgres_connector/ent/db"
+	db_credit "github.com/openmeterio/openmeter/internal/credit/postgres_connector/ent/db/creditentry"
+	credit_model "github.com/openmeterio/openmeter/pkg/credit"
+)
+
+// GetHighWatermark returns the high watermark for the given credit and subject pair.
+func (c *PostgresConnector) GetHighWatermark(ctx context.Context, namespace string, subject string) (credit_model.HighWatermark, error) {
+	lastReset, err := c.db.CreditEntry.Query().
+		Where(
+			db_credit.Namespace(namespace),
+			db_credit.Subject(subject),
+			db_credit.EntryTypeEQ(credit_model.EntryTypeReset),
```

Comment:
You can consider storing the high watermark so that we don't have to do a lookup in a possibly huge table (and we could put that into the table I proposed for locking).